drivers/test.go

```diff
@@ -144,6 +151,18 @@ func init() {
 		DefaultSystemActorState,
 		DefaultCronActorState,
 	}
+
+	var err error
+	PuppetActorAddress, err = address.NewIDAddress(98)
+	if err != nil {
+		panic(err)
+	}
+	PuppetActorState = ActorState{
+		Addr:    PuppetActorAddress,
+		Balance: abi_spec.NewTokenAmount(0),
+		Code:    puppet_spec.PuppetActorCodeID,
+		State:   &puppet_spec.State{},
+	}
```

Comment:
I can't see why the puppet actor is treated differently to the other ones. Please move this code up to 144 and follow the same pattern, including adding to DefaultBuiltinActorsState.

Alternatively, don't put any code here at all: defer it all to the caller of the builder. E.g. maybe they want a puppet with a balance? Maybe they want two puppets? This flexibility could be valuable tho we don't strictly need it now.

Either way is ok, but go all-in one or the other: builtin or not builtin.